### PR TITLE
Added Factfile as a parameter and in dockerfile

### PR DESCRIPTION
**Pull request by jonnaladheeraj**
Adding Factfile as a parameter when connected to puppet master it determines which modules needs to be installed depending on the facts file supplied as a parameter.

### CODE REVIEW

File: lib/puppet_x/puppetlabs/imagebuilder.rb [hunk 1]
```diff
@@ -163,6 +164,11 @@ def determine_if_using_puppetfile
         end
       end
 
+      def determine_if_using_factfile
+        if exists_and_is_file(:factfile)
+          @context[:use_factfile] = true
+        end
+       end
       def determine_if_using_hiera
```
Comment (sheenaajay) [inline lib/puppet_x/puppetlabs/imagebuilder.rb]: Add a new line after the function determine_if_using_factfile

File: templates/Dockerfile.erb [hunk 2]
```diff
@@ -47,6 +49,10 @@ RUN rm /usr/lib/ruby/gems/2.3.0/gems/facter-"$FACTER_VERSION"/lib/facter/blockde
 <% end %>
 <% end %>
 
+<% if use_factfile %>
+    COPY <%= factfile %> /etc/puppetlabs/facter/facts.d/custom_facts.txt
```
Comment (sheenaajay) [inline templates/Dockerfile.erb]: Just wondering if we should give an option to the user where to copy the files
/opt/puppetlabs/facter/facts.d/
/etc/puppetlabs/facter/facts.d/
/etc/facter/facts.d/

File: lib/puppet_x/puppetlabs/imagebuilder.rb (hunk 1 above)
Comment (jonnaladheeraj) [inline lib/puppet_x/puppetlabs/imagebuilder.rb]: I have added a new line after the function and made sure end is aligned with the function def

File: templates/Dockerfile.erb (hunk 2 above)
Comment (jonnaladheeraj) [inline templates/Dockerfile.erb]: Wanted all the facts to be at a location that are passed as a parameter. I can change it and ask user to enter the path for creation/checking facts.d folder and adding the facts file into it

**Comment by sheenaajay** [comment]
@jonnaladheeraj Thanks for the prompt reply.I committed some fixes for the travis CI build error.Could you please rebase them your branch to verify if travis CI build is passing.

**Comment by sheenaajay** [comment]
Travis erros are fixed and restarted the build succcesfully

**Pull request merged**